### ISSUE TITLE
Impossible equality outside conditional doesn't raise errors

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#838]
```php
$i = 5;
echo $i === null;
```
Expected: `int(5) does not contain null`
Actual: No issue - https://getpsalm.org/r/5d3a1564f6
